app/src/main/kotlin/com/hippo/ehviewer/ui/main/GalleryComment.kt

```diff
@@ -68,21 +65,46 @@ fun GalleryCommentCard(
                     },
                 )
             }
-            val textColor = LocalContentColor.current.toArgb()
-            val linkTextColor = MaterialTheme.colorScheme.primary.toArgb()
-            AndroidView(
-                factory = { context ->
-                    LinkifyTextView(context) { currentSpan = it }.apply {
-                        setTextColor(textColor)
-                        setLinkTextColor(linkTextColor)
-                    }
-                },
+            Column(
                 modifier = Modifier.constrainAs(commentRef) {
                     start.linkTo(parent.start)
                     top.linkTo(userRef.bottom, margin = 8.dp)
                 },
-                update = update,
-            )
+            ) {
+                ProvideTextStyle(value = MaterialTheme.typography.bodyMedium) {
+                    val linkColor = MaterialTheme.colorScheme.primary
+                    val processed = processComment(
+                        comment,
+                        TextLinkStyles(style = SpanStyle(color = linkColor)),
+                    ) { link ->
+                        check(link is LinkAnnotation.Url)
+                        onUrlClick(link.url)
+                    }
+                    val list = breakToTextAndUrl(comment.comment, processed)
+                    if (showImage) {
+                        list.forEach {
+                            when (it) {
+                                is Either.Left -> AsyncImage(
+                                    model = it.value,
+                                    contentDescription = null,
+                                )
+                                is Either.Right -> Text(text = it.value)
```

Comment:
Another thing to note is that text and images may be on the same line.

Reply:
Yeah, but how we handle that?

Reply:
Show image in another line doesn't make trouble, as image's height will never be same with text

Reply:
Untill some madlad put the image in the middle of the text

Reply:
We can't handle that easily as we need to measure text first.

Reply:
At least, if possible, let's do with another MR.

Reply:
Actually, I think we can use [`appendInlineContent`](https://developer.android.com/reference/kotlin/androidx/compose/ui/text/AnnotatedString.Builder#(androidx.compose.ui.text.AnnotatedString.Builder).appendInlineContent(kotlin.String,kotlin.String)) as we know the approximate size of the image (only thumbnails from e-hentai are allowed).

Reply:
How can we get approximate size w/o suspending?

Reply:
I don't think adding a 200x300 px placeholder is better than let image take a single line.

Reply:
Mostly it make sense only when image have about one or two line height

Reply:
Done some experiments and `appendInlineContent` has other issues.